.gitignore

```diff
@@ -257,8 +257,10 @@ cmake-build-debug
 # Below files are not deleted by "setup.py clean".
 
 # Visual Studio Code files
-.vscode
 .vs
+/.vscode/*
+!/.vscode/extensions.json
```

Comment:
`git clean -xfd` would still blow away vscode settings right? Maybe we should encourage people to use `python setup.py clean` instead

Reply:
it already does that even before this PR but yeah, that sounds like a good idea to me; in that case could you open a PR to replace `git clean -xfd` with `python setup.py clean` in `CONTRIBUTING.md`?